BlackBird proxy object error in Java 17

The object it's trying to serialize is an interface return value in a Spring REST controller, and the instance is constructed by a `@Query` in a Spring Repository

returned object looks like this
```java 
public interface GetStatusAccepted {
  Integer getAccepted();
  // more integer returning methods...
}
```

This problem was not occuring on identical code before we added BlackBirdModule to our Jackson configuration.

relevant part of trace:
`java.lang.IllegalAccessException: module jdk.proxy3 does not open jdk.proxy3 to unnamed module @479f2dc2
	at java.base/java.lang.invoke.MethodHandles.privateLookupIn(MethodHandles.java:259)
	at com.fasterxml.jackson.module.blackbird.util.ReflectionHack$Java9Up.lambda$privateLookupIn$0(ReflectionHack.java:53)
	at com.fasterxml.jackson.module.blackbird.util.Unchecked.lambda$supplier$1(Unchecked.java:41)
	at com.fasterxml.jackson.module.blackbird.util.ReflectionHack$Java9Up.privateLookupIn(ReflectionHack.java:53)
	at com.fasterxml.jackson.module.blackbird.util.ReflectionHack.privateLookupIn(ReflectionHack.java:25)
	at com.fasterxml.jackson.module.blackbird.ser.BBSerializerModifier.lambda$findProperties$0(BBSerializerModifier.java:67)
	at com.fasterxml.jackson.module.blackbird.util.Unchecked.lambda$runnable$0(Unchecked.java:31)
	at com.fasterxml.jackson.module.blackbird.ser.BBSerializerModifier.findProperties(BBSerializerModifier.java:68)
	at com.fasterxml.jackson.module.blackbird.ser.BBSerializerModifier.changeProperties(BBSerializerModifier.java:52)
	at com.fasterxml.jackson.databind.ser.BeanSerializerFactory.constructBeanOrAddOnSerializer(BeanSerializerFactory.java:414)
	at com.fasterxml.jackson.databind.ser.BeanSerializerFactory.findBeanOrAddOnSerializer(BeanSerializerFactory.java:294)
	at com.fasterxml.jackson.databind.ser.BeanSerializerFactory._createSerializer2(BeanSerializerFactory.java:239)
	at com.fasterxml.jackson.databind.ser.BeanSerializerFactory.createSerializer(BeanSerializerFactory.java:173)
	at com.fasterxml.jackson.databind.SerializerProvider._createUntypedSerializer(SerializerProvider.java:1495)
	at com.fasterxml.jackson.databind.SerializerProvider._createAndCacheUntypedSerializer(SerializerProvider.java:1443)
	at com.fasterxml.jackson.databind.SerializerProvider._findExplicitUntypedSerializer(SerializerProvider.java:1412)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.hasSerializerFor(DefaultSerializerProvider.java:260)
	at com.fasterxml.jackson.databind.ObjectMapper.canSerialize(ObjectMapper.java:3431)
	at org.springframework.http.converter.json.AbstractJackson2HttpMessageConverter.canWrite(AbstractJackson2HttpMessageConverter.java:274)
	at org.springframework.http.converter.AbstractGenericHttpMessageConverter.canWrite(AbstractGenericHttpMessageConverter.java:76)
	at org.springframework.web.servlet.mvc.method.annotation.AbstractMessageConverterMethodProcessor.getProducibleMediaTypes(AbstractMessageConverterMethodProcessor.java:380)
	at org.springframework.web.servlet.mvc.method.annotation.AbstractMessageConverterMethodProcessor.writeWithMessageConverters(AbstractMessageConverterMethodProcessor.java:230)
	at org.springframework.web.servlet.mvc.method.annotation.RequestResponseBodyMethodProcessor.handleReturnValue(RequestResponseBodyMethodProcessor.java:183)
	at org.springframework.web.method.support.HandlerMethodReturnValueHandlerComposite.handleReturnValue(HandlerMethodReturnValueHandlerComposite.java:78)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:124)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:895)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1064)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:963)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:898)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:645)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
`

I was able to create a test case that fails:
```
public class JacksonMB181Test {
    @Test
    public void wholeEnchilada() throws Exception {
        final var om = new ObjectMapper();
        om.registerModule(new BlackbirdModule());
        final var val = om.writeValueAsString(Proxy.newProxyInstance(JacksonBBTest.class.getClassLoader(), new Class<?>[] { Beany.class }, (p, m, a) -> {
            if (m.getName().equals("a")) {
                return 42;
            }
            return null;
        }));
        assertThat(val).isEqualTo("{\"a\":42}");
    }

    public interface Beany {
        int getA();
    }
}
```

```
java.lang.RuntimeException: java.lang.IllegalAccessException: module jdk.proxy2 does not open jdk.proxy2 to unnamed module @2d4e282b
	at com.fasterxml.jackson.databind.util.ClassUtil.closeOnFailAndThrowAsIOE(ClassUtil.java:502)
	at com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4626)
	at com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:3869)
	at com.paywholesail.service.api.JacksonBBTest.wholeEnchilada(JacksonBBTest.java:16)
Caused by: java.lang.IllegalAccessException: module jdk.proxy2 does not open jdk.proxy2 to unnamed module @2d4e282b
	at java.base/java.lang.invoke.MethodHandles.privateLookupIn(MethodHandles.java:279)
	at com.fasterxml.jackson.module.blackbird.util.ReflectionHack$Java9Up.lambda$privateLookupIn$0(ReflectionHack.java:55)
	at com.fasterxml.jackson.module.blackbird.util.Unchecked.lambda$supplier$1(Unchecked.java:41)
	at com.fasterxml.jackson.module.blackbird.util.ReflectionHack$Java9Up.privateLookupIn(ReflectionHack.java:55)
	at com.fasterxml.jackson.module.blackbird.util.ReflectionHack.privateLookupIn(ReflectionHack.java:25)
	at com.fasterxml.jackson.module.blackbird.ser.BBSerializerModifier.lambda$findProperties$0(BBSerializerModifier.java:67)
	at com.fasterxml.jackson.module.blackbird.util.Unchecked.lambda$runnable$0(Unchecked.java:31)
	at com.fasterxml.jackson.module.blackbird.ser.BBSerializerModifier.findProperties(BBSerializerModifier.java:68)
	at com.fasterxml.jackson.module.blackbird.ser.BBSerializerModifier.changeProperties(BBSerializerModifier.java:52)
	at com.fasterxml.jackson.databind.ser.BeanSerializerFactory.constructBeanOrAddOnSerializer(BeanSerializerFactory.java:415)
	at com.fasterxml.jackson.databind.ser.BeanSerializerFactory.findBeanOrAddOnSerializer(BeanSerializerFactory.java:295)
	at com.fasterxml.jackson.databind.ser.BeanSerializerFactory._createSerializer2(BeanSerializerFactory.java:240)
	at com.fasterxml.jackson.databind.ser.BeanSerializerFactory.createSerializer(BeanSerializerFactory.java:174)
	at com.fasterxml.jackson.databind.SerializerProvider._createUntypedSerializer(SerializerProvider.java:1501)
	at com.fasterxml.jackson.databind.SerializerProvider._createAndCacheUntypedSerializer(SerializerProvider.java:1449)
	at com.fasterxml.jackson.databind.SerializerProvider.findValueSerializer(SerializerProvider.java:550)
	at com.fasterxml.jackson.databind.SerializerProvider.findTypedValueSerializer(SerializerProvider.java:828)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:308)
	at com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4624)
	... 26 more
```

--- a/blackbird/src/main/java/com/fasterxml/jackson/module/blackbird/util/ReflectionHack.java
+++ b/blackbird/src/main/java/com/fasterxml/jackson/module/blackbird/util/ReflectionHack.java
@@ -52,7 +52,13 @@ public class ReflectionHack {
         }
 
         public static Lookup privateLookupIn(Class<?> lookup, Lookup orig) {
-            return Unchecked.supplier(() -> (Lookup) FACTORY.invokeExact(lookup, orig)).get();
+            try {
+                return (Lookup) FACTORY.invokeExact(lookup, orig);
+            } catch (ReflectiveOperationException e) {
+                return orig;
+            } catch (Throwable t) {
+                throw Sneaky.throwAnyway(t);
+            }
         }
     }
 

--- a/blackbird/src/test/java/com/fasterxml/jackson/module/blackbird/TestAccessFallback.java
+++ b/blackbird/src/test/java/com/fasterxml/jackson/module/blackbird/TestAccessFallback.java
@@ -1,5 +1,7 @@
 package com.fasterxml.jackson.module.blackbird;
 
+import java.lang.reflect.Proxy;
+
 import com.fasterxml.jackson.databind.ObjectMapper;
 
 public class TestAccessFallback extends BlackbirdTestBase
@@ -68,5 +70,20 @@ public class TestAccessFallback extends BlackbirdTestBase
         // actually try again, to ensure handling works reliably
         MyBean bean2 = abMapper.readValue(BEAN_JSON, MyBean.class);
         assertEquals("a", bean2.getE());
+    }
+
+    public void testProxyAccessIssue181() throws Exception {
+        ObjectMapper om = newObjectMapper();
+        String val = om.writeValueAsString(Proxy.newProxyInstance(TestAccessFallback.class.getClassLoader(), new Class<?>[] { Beany.class }, (p, m, a) -> {
+            if (m.getName().equals("getA")) {
+                return 42;
+            }
+            return null;
+        }));
+        assertEquals("{\"a\":42}", val);
+    }
+
+    public interface Beany {
+        int getA();
     }
 }

